doc/man1/kdf.pod

```diff
@@ -0,0 +1,157 @@
+=pod
+
+=head1 NAME
+
+openssl-kdf,
+kdf - perform Key Derivation Function operations
+
+=head1 SYNOPSIS
+
+B<openssl kdf>
+[B<-help>]
+[B<-kdfopt>]
+[B<-keylen num>]
+[B<-out filename>]
+[B<-binary>]
+B<kdf_name>
+
+B<openssl> I<kdf> [B<...>] B<kdf_name>
+
+=head1 DESCRIPTION
+
+The key derivation functions generate a derived key from either a secret OR
+password.
+
+=head1 OPTIONS
+
+=over 4
+
+=item B<-help>
+
+Print a usage message.
+
+=item B<-keylen num>
+
+The output size of the derived key. This field is required.
+
+=item B<-out filename>
+
+Filename to output to, or standard output by default.
+
+=item B<-binary>
+
+Output the derived key in binary form. Uses hexadecimal text format if not specified.
+
+=item B<-kdfopt nm:v>
+
+Passes options to the KDF algorithm.
+A comprehensive list of controls can be found in the EVP_KDF_CTX implementation
+documentation.
+Common control strings used by EVP_KDF_ctrl_str() are:
+
+=over 4
+
+=item B<key:string>
+
+Specifies the secret key as an alphanumeric string (use if the key contains
+printable characters only).
+The string length must conform to any restrictions of the KDF algorithm.
+A key must be specified for most KDF algorithms.
+
+=item B<hexkey:string>
+
+Specifies the secret key in hexadecimal form (two hex digits per byte).
+The key length must conform to any restrictions of the KDF algorithm.
+A key must be specified for most KDF algorithms.
+
+=item B<password:string>
+
+Specifies the password as an alphanumeric string (use if the password contains
+printable characters only).
+The password must be specified for most KDF algorithms.
+
+=item B<password:string>
+
+Specifies the secret key in hexadecimal form (two hex digits per byte).
+The key length must conform to any restrictions of the KDF algorithm.
+A key must be specified for most KDF algorithms.
+
+=item B<digest:string>
+
+Used by TLS1-PRF, HKDF, SSKDF HMAC, SSKDF Hash, and PBKDF2, as an alphanumeric
+string.
+To see the list of supported digests, use the command I<list -digest-commands>.
+
+=back
+
+=item B<kdf_name>
+
+Specifies the name of a supported KDF algorithm which will be used.
+The supported algorithms are TLS1-PRF, HKDF, SSKDF, PBKDF2 and id-scrypt.
+
+=back
+
+=head1 EXAMPLES
+
+Use TLS1-PRF to create a hex-encoded derived key from a secret key and seed: \
+openssl kdf -keylen 16 -kdfopt digest:SHA256 -kdfopt key:secret \
+        -kdfopt seed:seed TLS1-PRF
```

Comment:
Not sure I understand what you want to do here...  I think the section should be formatted like this:

```
Use TLS1-PRF to create a hex-encoded derived key from a secret key and seed:

    openssl kdf -keylen 16 -kdfopt digest:SHA256 -kdfopt key:secret \
            -kdfopt seed:seed TLS1-PRF
```

Reply:
... and similarly for all other examples

Reply:
Hmm still looks bad.. I changed it to this.

=head2 Use TLS1-PRF to create a hex-encoded derived key from a secret key and seed:
C<openssl kdf -keylen 16 -kdfopt digest:SHA256 -kdfopt key:secret -kdfopt seed:seed TLS1-PRF>